doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -5,269 +5,297 @@ Reasoning with inductive types
+      :tacn:`exists`, :tacn:`split`, :tacn:`left` and :tacn:`right`,
+      but they introduce existential variables instead of failing
+      when a variable can't be instantiated
       (cf. :tacn:`eapply` and :tacn:`apply`).
 
-.. _CaseAnalysisAndInduction:
+.. example:: :tacn:`constructor`, :tacn:`left` and :tacn:`right`
 
-Case analysis and induction
--------------------------------
+   .. coqtop:: reset all
 
-The tactics presented in this section implement induction or case
-analysis on inductive or co-inductive objects (see :ref:`inductive-definitions`).
+      Print or.  (* or, represented by \/, has two constructors, or_introl and or_intror *)
+      Goal  forall P1 P2 : Prop, P1 -> P1 \/ P2.
+      constructor 1.  (* equivalent to "left" *)
+      apply H.  (* success *)
 
-.. tacn:: destruct @term
-   :name: destruct
+   In contrast, we won't be able to complete the proof if we select constructor 2:
 
-   This tactic applies to any goal. The argument :token:`term` must be of
-   inductive or co-inductive type and the tactic generates subgoals, one
-   for each possible form of :token:`term`, i.e. one for each constructor of the
-   inductive or co-inductive type. Unlike :tacn:`induction`, no induction
-   hypothesis is generated by :tacn:`destruct`.
+   .. coqtop:: reset none
 
-   .. tacv:: destruct @ident
+      Goal  forall P1 P2 : Prop, P1 -> P1 \/ P2.
 
-      If :token:`ident` denotes a quantified variable of the conclusion
-      of the goal, then :n:`destruct @ident` behaves
-      as :n:`intros until @ident; destruct @ident`. If :token:`ident` is not
-      anymore dependent in the goal after application of :tacn:`destruct`, it
-      is erased (to avoid erasure, use parentheses, as in :n:`destruct (@ident)`).
+   .. coqtop:: all
 
-      If :token:`ident` is a hypothesis of the context, and :token:`ident`
-      is not anymore dependent in the goal after application
-      of :tacn:`destruct`, it is erased (to avoid erasure, use parentheses, as
-      in :n:`destruct (@ident)`).
+      constructor 2.  (* equivalent to "right" *)
 
-   .. tacv:: destruct @natural
+   You can also apply a constructor by name.  (`apply or_intror` will fail):
 
-      :n:`destruct @natural` behaves as :n:`intros until @natural`
-      followed by destruct applied to the last introduced hypothesis.
+   .. coqtop:: reset none
 
-     .. note::
-        For destruction of a number, use syntax :n:`destruct (@natural)` (not
-        very interesting anyway).
+      Goal  forall P1 P2 : Prop, P1 -> P1 \/ P2.
 
-   .. tacv:: destruct @pattern
+   .. coqtop:: all
 
-      The argument of :tacn:`destruct` can also be a pattern of which holes are
-      denoted by “_”. In this case, the tactic checks that all subterms
-      matching the pattern in the conclusion and the hypotheses are compatible
-      and performs case analysis using this subterm.
+      apply or_introl.  (* equivalent to "left" *)
```

Comment:
```suggestion
      intros; apply or_introl.  (* equivalent to "left" *)
```

If you do not do the `intros`, then it is not equivalent to `constructor 1` / `left`.

With the `intros`, it doesn't solve the goal and `apply or_intror` doesn't fail (it just leads to a dead-end).